src/App.jsx

```diff
@@ -1,45 +1,61 @@
-import React, { useState } from 'react';
+import React, { useState, useEffect } from 'react';
+import { Routes, Route, Navigate, useNavigate, useLocation } from 'react-router-dom';
 import Header from './components/Header/Header';
 import Courses from './components/Courses/Courses';
 import CreateCourse from './components/CreateCourse/CreateCourse';
+import Registration from './components/Registration/Registration';
+import Login from './components/Login/Login';
 import './App.css';
 import { mockedCoursesList, mockedAuthorsList } from './constants';
 
 const App = () => {
   const [courses, setCourses] = useState(mockedCoursesList);
   const [authors, setAuthors] = useState(mockedAuthorsList);
-  const [isCreatingCourse, setIsCreatingCourse] = useState(false);
+  const [userName, setUserName] = useState(localStorage.getItem('userName') || '');
```

Comment:
It's a good practice to put such strings ('userName', 'userToken', 'login', 'registration') into constants: https://d17btkcdsmqrmh.cloudfront.net/react-fundamentals/docs/module-1/home-task/good_practices #4.